python/raydp/tf/estimator.py

```diff
@@ -240,16 +241,27 @@ def fit_on_spark(self,
                      stop_spark_after_conversion=False) -> NoReturn:
         super().fit_on_spark(train_df, evaluate_df)
         train_df = self._check_and_convert(train_df)
-        train_ds = spark_dataframe_to_ray_dataset(train_df,
+        if fs_directory is not None:
+            train_df.write.parquet(fs_directory, compression=compression)
+            train_ds = ray.data.read_parquet(fs_directory)
+        else:
+            train_ds = spark_dataframe_to_ray_dataset(train_df,
+                                                  parallelism=self._num_workers,
```

Comment:
@KepingYan previously removed the parallelism on purpose for better performance.